packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -0,0 +1,509 @@
+  }
+
+  public readonly clusterIdentifier: string;
+  public readonly clusterEndpoint: Endpoint;
+  public readonly clusterReadEndpoint: Endpoint;
+  public readonly connections: ec2.Connections;
+
+  /**
+   * The secret attached to this cluster
+   */
+  private readonly secret?: secretsmanager.ISecret;
+  private readonly subnetGroup: ISubnetGroup;
+  private readonly vpc: ec2.IVpc;
+  private readonly vpcSubnets?: ec2.SubnetSelection;
+
+  private readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+  private readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  constructor(scope:Construct, id: string, props: ServerlessClusterProps) {
+    super(scope, id);
+
+    this.vpc = props.vpc;
+    this.vpcSubnets = props.vpcSubnets;
+
+    this.singleUserRotationApplication = props.engine.singleUserRotationApplication;
+    this.multiUserRotationApplication = props.engine.multiUserRotationApplication;
+
+    const { subnetIds } = this.vpc.selectSubnets(this.vpcSubnets);
+
+    // Cannot test whether the subnets are in different AZs, but at least we can test the amount.
+    if (subnetIds.length < 2) {
+      Annotations.of(this).addError(`Cluster requires at least 2 subnets, got ${subnetIds.length}`);
+    }
+
+    this.subnetGroup = props.subnetGroup ?? new SubnetGroup(this, 'Subnets', {
+      description: `Subnets for ${id} database`,
+      vpc: props.vpc,
+      vpcSubnets: props.vpcSubnets,
+      removalPolicy: props.removalPolicy === RemovalPolicy.RETAIN ? props.removalPolicy : undefined,
+    });
+
+    let credentials = props.credentials ?? Credentials.fromUsername(props.engine.defaultUsername ?? 'admin');
+    if (!credentials.secret && !credentials.password) {
+      credentials = Credentials.fromSecret(new DatabaseSecret(this, 'Secret', {
+        username: credentials.username,
+        encryptionKey: credentials.encryptionKey,
+      }));
+    }
+    const secret = credentials.secret;
+
+    // bind the engine to the Cluster
+    const clusterEngineBindConfig = props.engine.bindToCluster(this, {
+      parameterGroup: props.parameterGroup,
+    });
+    const clusterParameterGroup = props.parameterGroup ?? clusterEngineBindConfig.parameterGroup;
+    const clusterParameterGroupConfig = clusterParameterGroup?.bindToCluster({});
+
+    const securityGroups = props.securityGroups ?? [
+      new ec2.SecurityGroup(this, 'SecurityGroup', {
+        description: 'RDS security group',
+        vpc: this.vpc,
+      }),
+    ];
+
+    const cluster = new CfnDBCluster(this, 'Resource', {
+      backupRetentionPeriod: props.backupRetention?.toDays(),
+      databaseName: props.defaultDatabaseName,
+      dbClusterIdentifier: props.clusterIdentifier,
+      dbClusterParameterGroupName: clusterParameterGroupConfig?.parameterGroupName,
+      dbSubnetGroupName: this.subnetGroup.subnetGroupName,
+      deletionProtection: props.deletionProtection,
```

Comment:
This is incorrect (should be [this](https://github.com/aws/aws-cdk/blob/8a0401495e05e8ec7178d2ac57556434c853cabd/packages/%40aws-cdk/aws-rds/lib/cluster.ts#L353-L353)).

Reply:
yep, was integrating with the refactor (had similar changes for the excludeCharacters which wasn't being leveraged in the serverless cluster either. also fixed that up).